common/src/KokkosKernels_SimpleUtils.hpp

```diff
@@ -380,6 +380,38 @@ KOKKOS_FORCEINLINE_FUNCTION Value xorshiftHash(Value v) {
              : static_cast<Value>(x * 2685821657736338717ULL - 1);
 }
 
+struct ViewHashFunctor {
+  ViewHashFunctor(const uint8_t *data_) : data(data_) {}
+
+  KOKKOS_INLINE_FUNCTION void operator()(size_t i, uint32_t &lhash) const {
+    // Compute a hash/digest of both the index i, and data[i]. Then add that to
+    // overall hash.
+    uint32_t x = uint32_t(i);
+    x ^= x << 13;
+    x ^= x >> 17;
+    x ^= x << 5;
+    x ^= uint32_t(data[i]);
```

Comment:
Is this safe if value_type is a struct with padding between members?

Reply:
Perhaps in spgemm debug mode, this padding will be initialized to 0?

Reply:
Hmm, that is a good point - I don't think this work (if the padding is left uninitialized, hashes for views that should be identical could come out different). I will try to find a way to static_assert that there is no padding, there's probably a way to do that in the standard library somewhere.

Reply:
Aha, ``std::has_unique_object_representations<T>`` says this:

Notes: This trait was introduced to make it possible to determine whether a type can be correctly hashed by hashing its object representation as a byte array.

Reply:
It now checks at compile-time that value_type has no padding bytes.